src/components/DayPickerSingleDateController.jsx

```diff
@@ -0,0 +1,572 @@
+
+  isRTL: false,
+};
+
+export default class DayPickerSingleDateController extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.isTouchDevice = false;
+    this.today = moment();
+
+    this.modifiers = {
+      today: day => this.isToday(day),
+      blocked: day => this.isBlocked(day),
+      'blocked-calendar': day => props.isDayBlocked(day),
+      'blocked-out-of-range': day => props.isOutsideRange(day),
+      'highlighted-calendar': day => props.isDayHighlighted(day),
+      valid: day => !this.isBlocked(day),
+      hovered: day => this.isHovered(day),
+      selected: day => this.isSelected(day),
+    };
+
+    const { currentMonth, visibleDays } = this.getStateForNewMonth(props);
+
+    this.state = {
+      hoverDate: null,
+      currentMonth,
+      visibleDays,
+    };
+
+    this.onDayMouseEnter = this.onDayMouseEnter.bind(this);
+    this.onDayMouseLeave = this.onDayMouseLeave.bind(this);
+    this.onDayClick = this.onDayClick.bind(this);
+
+    this.onPrevMonthClick = this.onPrevMonthClick.bind(this);
+    this.onNextMonthClick = this.onNextMonthClick.bind(this);
+
+    this.getFirstFocusableDay = this.getFirstFocusableDay.bind(this);
+  }
+
+  componentDidMount() {
+    this.isTouchDevice = isTouchDevice();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const {
+      date,
+      focused,
+      isOutsideRange,
+      isDayBlocked,
+      isDayHighlighted,
+      initialVisibleMonth,
+      numberOfMonths,
+      enableOutsideDays,
+    } = nextProps;
+    let { visibleDays } = this.state;
+
+    if (isOutsideRange !== this.props.isOutsideRange) {
+      this.modifiers['blocked-out-of-range'] = day => isOutsideRange(day);
+    }
+
+    if (isDayBlocked !== this.props.isDayBlocked) {
+      this.modifiers['blocked-calendar'] = day => isDayBlocked(day);
+    }
+
+    if (isDayHighlighted !== this.props.isDayHighlighted) {
+      this.modifiers['highlighted-calendar'] = day => isDayHighlighted(day);
+    }
+
+    if (
+      initialVisibleMonth !== this.props.initialVisibleMonth ||
+      numberOfMonths !== this.props.numberOfMonths ||
+      enableOutsideDays !== this.props.enableOutsideDays
+    ) {
+      const newMonthState = this.getStateForNewMonth(nextProps);
+      const currentMonth = newMonthState.currentMonth;
+      visibleDays = newMonthState.visibleDays;
+      this.setState({
+        currentMonth,
+        visibleDays,
+      });
+    }
+
+    const didDateChange = date !== this.props.date;
+    const didFocusChange = focused !== this.props.focused;
+
+    let modifiers = {};
+
+    if (didDateChange) {
+      modifiers = this.deleteModifier(modifiers, this.props.date, 'selected');
+      modifiers = this.addModifier(modifiers, date, 'selected');
+    }
+
+    if (didFocusChange) {
+      values(visibleDays).forEach((days) => {
+        Object.keys(days).forEach((day) => {
+          const momentObj = moment(day);
+          if (isDayBlocked(momentObj)) {
```

Comment:
@majapw I think this lost the change fixed in #550. Specifically, checks for `isBlocked` and `isOutsideRange`. Cheers!